giskard/testing/tests/miscellaneous.py

```diff
@@ -0,0 +1,234 @@
+from typing import List, Optional
+
+import numpy as np
+
+from giskard import Dataset, TestResult, test
+from giskard.models.base import BaseModel
+from giskard.testing.tests.debug_slicing_functions import row_failing_slicing_fn
+
+
+def _check_columns(column_names, column_values):
+    if column_values is None:
+        return
+
+    if set(column_names) != set(column_values.keys()):
+        raise ValueError("column_values keys should be the same as column_names")
+
+    lengths = [len(elt) for elt in list(column_values.values())]
+    if not all(x == lengths[0] for x in lengths):
+        raise ValueError("column_values must contain elements with same length")
+
+
+def _get_predictions(
+    model: BaseModel,
+    dataset: Dataset,
+    column_names: List[str],
+    column_values: Optional[dict] = None,
+    random_state: int = 123456,
+    num_samples: int = 100,
+    num_grid: int = 50,
+    classif_index_label: int = 0,
+):
+    # Create grids of values per column if not passed as input
+    if column_values is None:
+        column_values = {}
+        for col_name in column_names:
+            min_col = dataset.df[col_name].min()
+            max_col = dataset.df[col_name].max()
+
+            # Create grid of values between min and max
+            column_values[col_name] = np.linspace(min_col, max_col, num_grid)
+    else:
+        num_grid = len(list(column_values.values())[0])
+
+    # Generate data from existing rows by modifying the value of the column of interest
+    augmented_df = []
+    sample_rows = dataset.df.sample(n=num_samples, random_state=random_state).copy()
+    for i in range(num_grid):
+        sample_rows_copy = sample_rows.copy()
+        for col_name in column_names:
+            sample_rows_copy.loc[:, col_name] = column_values[col_name][i].astype(dataset.df.dtypes[col_name])
+        augmented_df.append(
+            Dataset(
+                sample_rows_copy,
+                target=dataset.target,
+                cat_columns=dataset.cat_columns,
+                column_types=dataset.column_types,
+                validation=False,
+            )
+        )
```

Comment:
I don't understand what's going on here. 

Why are you creating a `num_grid` number of Datasets? Why are you casing each time the types?

To me what makes sense would be:
```py
for feat_name in feature_names:
  sample_rows_copy = sample_rows.copy()
  sample_rows_copy.loc[:, feat_name] = column_values[feat_name].astype(dataset.df.dtypes[feat_name])
  augmented_ds.append(
            Dataset(
                sample_rows_copy,
                target=dataset.target,
                cat_columns=dataset.cat_columns,
                column_types=dataset.column_types,
                validation=False,
            )
        )
```

what am i missing?